index.ts

```diff
@@ -1,30 +1,50 @@
-// @flow
-
 import GeocoderArcGIS from 'geocoder-arcgis'
-import lonlat from '@conveyal/lonlat'
+import type { GeoJSON } from 'geojson'
+import { toCoordinates, toString, toPoint } from '@conveyal/lonlat'
+import type { LonLatInput } from '@conveyal/lonlat'
 
-const arcGisGeocoders = {}
+type Point = {
+  x: number
+  y: number
+}
+
+type GeocodersMap = {
+  [key: string]: GeocoderArcGIS
+}
 
 type Boundary = {
   rect: {
-    minLat: number,
-    minLon: number,
-    maxLat: number,
+    minLat: number
+    minLon: number
+    maxLat: number
     maxLon: number
   }
 }
 
+type Candidate = {
+  location: LonLatInput
+  attributes: Record<string, string | number>
+}
+
 type BaseQuery = {
-  clientId?: string,
-  clientSecret?: string,
+  clientId?: string
+  clientSecret?: string
   url?: string
 }
 
-type BaseResponse = {
-  query: any
+type GeocoderResponse = {
+  suggestions?: string[]
+  locations?: Candidate[]
+  location?: LonLatInput
+  address?: Record<string, string>
+  candidates?: Candidate[]
 }
```

Comment:
This is too generic of a type to use for the parse results of each geocoding response. I recommend creating a specific response type for each API call (`AutocompleteResponse`, `BulkResopsne`, `ReverseResponse`, `SearchResponse`).

Reply:
All of them were inline-able! Good recommendation thank you. Fixed in cdd90839c90a85888732fb6718964b8ddd8fb1b6